src/test/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/action/ForceMergeActionIT.kt

```diff
@@ -0,0 +1,222 @@
+                listOf(forceMergeActionConfig),
+                listOf(Transition(stateName = "DeleteState", conditions = null))
+            ),
+            State("DeleteState", listOf(deleteActionConfig), listOf())
+        )
+
+        val policy = Policy(
+            id = policyID,
+            description = "$testIndexName description",
+            schemaVersion = 1L,
+            lastUpdatedTime = Instant.now().truncatedTo(ChronoUnit.MILLIS),
+            defaultNotification = randomDefaultNotification(),
+            defaultState = states[0].name,
+            states = states
+        )
+
+        createPolicy(policy, policyID)
+        createIndex(indexName, null)
+        refresh()
+
+        // Add sample data to increase segment count, passing in a delay so that multiple segments get created
+        insertSampleData(indexName, 3, 1000)
+
+        val segmentCount = getSegmentCount(indexName)
+        assertTrue("Segment count for [$indexName] was less than expected", segmentCount > 1)
+
+        // Add policy to index after data is loaded
+        addPolicyToIndex(indexName, policyID)
+        Thread.sleep(2000)
+
+        val managedIndexConfig = getManagedIndexConfig(indexName)
+        assertNotNull("ManagedIndexConfig is null", managedIndexConfig)
+
+        // Will change the startTime each execution so that it triggers in 2 seconds
+        // First execution: Policy is initialized
+        updateManagedIndexConfigStartTime(managedIndexConfig!!, Instant.now().minusSeconds(58).toEpochMilli())
+        Thread.sleep(3000)
+
+        // Second execution: Index is set to read-only for force_merge
+        updateManagedIndexConfigStartTime(managedIndexConfig, Instant.now().minusSeconds(58).toEpochMilli())
+        Thread.sleep(3000)
+
+        assertEquals("true", getIndexBlocksWriteSetting(indexName))
+
+        // Third execution: Force merge operation is kicked off
+        updateManagedIndexConfigStartTime(managedIndexConfig, Instant.now().minusSeconds(58).toEpochMilli())
+        Thread.sleep(3000)
+
+        // Fourth execution: Waits for force merge to complete, which will happen in this execution since index is small
+        updateManagedIndexConfigStartTime(managedIndexConfig, Instant.now().minusSeconds(58).toEpochMilli())
+        Thread.sleep(3000)
+
+        val segmentCountAfterForceMerge = getSegmentCount(indexName)
+        assertEquals("Segment count for [$indexName] after force merge is incorrect", 1, segmentCountAfterForceMerge)
+
+        // Fifth execution: Set index back to read-write since it was not originally read-only
+        updateManagedIndexConfigStartTime(managedIndexConfig, Instant.now().minusSeconds(58).toEpochMilli())
+        Thread.sleep(3000)
+
+        assertEquals("false", getIndexBlocksWriteSetting(indexName))
+
+        // Sixth execution: Transition to DeleteState
+        updateManagedIndexConfigStartTime(managedIndexConfig, Instant.now().minusSeconds(58).toEpochMilli())
```

Comment:
No need to test delete, we already test it in its own IT, should be fine to stop after asserting index set back to read-write